artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPLargeMessage.java

```diff
@@ -126,6 +127,20 @@ public void closeLargeMessage() throws Exception {
       parsingData = null;
    }
 
+   public synchronized void releaseEncodedBuffer() {
```

Comment:
Why synchronized? We perform these operations concurrently? 
If the semantic of accesses is as-single-threaded, but is happening on different Threads one-by-one, is ok to remove the synchronization (and better, because better document that this class shouldn't be used for concurrent accesses)

Reply:
Largemessage can be concurrent accessed and used. In case of topic with multiple queues. 

Remember how we had to fix loads of concurrent access issue on core message the other year

Reply:
Id rather this is sync. Dont need to be chasing such difficult issues again

Reply:
Yep, agree I just wanted be sure was a correctness requirement :)

Reply:
This is about persistence.  Which will not happen concurrently unless it’s replication.

Reply:
So, I will keep the synchronize here.